headers/components/temperature_component.hpp

```diff
@@ -1,87 +0,0 @@
-/* Hector -- A Simple Climate Model
```

Comment:
I really don't love that we are deleting the original temperature component entirely.  The whole idea behind Hector's design is that we are supposed to be able to add and drop components as needed.  I get that we don't want components to proliferate because then we have to maintain them all, but so far maintaining components hasn't been a huge amount of effort.  Is the original temperature component really unable to coexist with the DOEClim temperature (in the code base, I mean; obviously only one of them can be active in any single run)?

Reply:
In my experience, it works fine with both components and just specifying which version you want to use in the .ini file. It does lead to a couple warnings like below:
WARNING:registerCapability: temperature is declaring capability Tgav previously registered

Reply:
They can exist together and do currently.  My concern is that the old way was a hack with a bunch of bugs that were found. I don't think I want people having an easy option to get back incorrect results by turning on that old component.

Reply:
That sounds reasonable.  Since that leaves the doeclim temperature component as the _only_ temperature component, maybe we should just rename it to "temperature_component".  (We would still explain its doeclim origins in comments in the source file, of course.)

Reply:
done